File: src/cviz/state/State.java
package cviz.state;

import cviz.TimelineState;
import cviz.control.IControlInterface;

import java.io.Serializable;

public class State implements Serializable{
    private transient final IControlInterface controlInterface;

    private final String timelineSlot;
    private final String timelineFile;
    private final String instanceName;
    private TimelineState state;
    private String stateMessage;

    public State(IControlInterface controlInterface, String timelineSlot, String timelineFile, String instanceName){
        this.controlInterface = controlInterface;
        this.timelineSlot = timelineSlot;
        this.timelineFile = timelineFile;

        this.instanceName = instanceName;

        this.state = TimelineState.READY;
        this.stateMessage = null;
    }

    public void setState(TimelineState state){
        setState(state, null);
    }
    public void setState(TimelineState state, String stateMessage){
        this.state = state;
        this.stateMessage = stateMessage;

        controlInterface.notifyState(this);
    }

    public String getTimelineSlot() {
        return timelineSlot;
    }

    public String getTimelineFile() {
        return timelineFile;
    }

    public String getInstanceName() {
        return instanceName;
    }

    public TimelineState getState() {
        return state;
    }

    public String getStateMessage() {
        return stateMessage;
    }
}

File: CViz/src/State/State.cs
using System;
using CViz.Control;

namespace CViz.State
{
    class State
    {
        private readonly IControlInterface _controlInterface;

        public State(IControlInterface controlInterface, string timelineSlot, string timelineFile, string instanceName)
        {
            _controlInterface = controlInterface;
            TimelineSlot = timelineSlot;
            TimelineFile = timelineFile;
            InstanceName = instanceName;

            TimelineState = TimelineState.Ready;
            StateMessage = null;
        }

        public string TimelineSlot { get; }
        public string TimelineFile { get; }
        public string InstanceName { get; }

        public TimelineState TimelineState { get; private set; }
        public string StateMessage { get; private set; }

        public void SetState(TimelineState state, string message = null)
        {
            TimelineState = state;
            StateMessage = message;

            _controlInterface.NotifyState(this);
        }
    }
}
